examples/typescript-step-functions-mixed/main.ts

```diff
@@ -0,0 +1,73 @@
+import { Construct } from "constructs";
+import { App, TerraformStack } from "cdktf";
+import {
+  Duration,
+  aws_stepfunctions,
+  aws_stepfunctions_tasks,
+  aws_lambda,
+} from "aws-cdk-lib";
+import * as aws_cdk from "aws-cdk-lib";
+import * as path from "path";
+
+import { AwsTerraformAdapter, AwsProvider } from "@cdktf/aws-cdk";
+import { NodejsFunction } from "@cdktf-plus/aws";
+
+export class StepFunctionsStack extends TerraformStack {
+  constructor(scope: Construct, name: string) {
+    super(scope, name);
+
+    new AwsProvider(this, "aws", { region: "us-west-2" });
+
+    const awsAdapter = new AwsTerraformAdapter(this, "adapter");
+
+    const functionGenerateID = new NodejsFunction(this, "generate-id", {
+      path: path.join(__dirname, "fn-generate-id.ts"),
+    });
+
+    const functionReverseID = new NodejsFunction(this, "reverse-id", {
+      path: path.join(__dirname, "fn-reverse-id.ts"),
+    });
+
+    const definition = new aws_stepfunctions_tasks.LambdaInvoke(
+      awsAdapter,
+      "Generate ID",
+      {
+        lambdaFunction: aws_lambda.Function.fromFunctionArn(
+          awsAdapter,
+          "generate-id",
+          aws_cdk.Lazy.string({
+            produce: () => functionGenerateID.fn.arn,
+          })
+        ),
+        outputPath: "$.Payload",
+      }
+    )
+      .next(
+        new aws_stepfunctions.Wait(awsAdapter, "Wait 1 Second", {
+          time: aws_stepfunctions.WaitTime.duration(Duration.seconds(1)),
+        })
+      )
+      .next(
+        new aws_stepfunctions_tasks.LambdaInvoke(awsAdapter, "Reverse ID", {
+          lambdaFunction: aws_lambda.Function.fromFunctionArn(
+            awsAdapter,
+            "reverse-id",
+            aws_cdk.Lazy.string({
+              produce: () => functionReverseID.fn.arn,
+            })
```

Comment:
That's awesome! Have you deployed this to AWS like that successfully?

Reply:
Yes! I did and the step function still worked 👍